### PR TITLE
refactor: vendir secrets

### PR DESCRIPTION
**Pull request by Zebradil**
The main change here is moving discovered vendir secrets from on-disk to in-memory.

~~As a next step, I want to move the process of collecting secrets into possession of the Globe, so that the discovery is done only once.~~ Done

### CODE REVIEW

File: internal/myks/globe.go [hunk 1]
```diff
@@ -373,3 +393,82 @@ func (g *Globe) Msg(msg string) string {
...
+		if err != nil {
+			return secretYamls, err
+		}
+		secretYamls += "---\n" + secretYaml
+	}
+
+	return secretYamls, nil
+}
+
+func generateVendirSecretYaml(secretName string, username string, password string) (string, error) {
```
Comment (fritzduchardt) [inline internal/myks/globe.go:457]:
```suggestion
func (g *Globe) generateVendirSecretYaml(secretName string, username string, password string) (string, error) {
```

File: internal/myks/globe.go [hunk 2]
```diff
@@ -373,3 +393,82 @@ func (g *Globe) Msg(msg string) string {
...
+
+	return vendirCredentials
+}
+
+func (g *Globe) generateVendirSecretYamls() (string, error) {
+	vendirCredentials := g.collectVendirSecrets()
+
+	var secretYamls string
+	for secretName, credentials := range vendirCredentials {
+		secretYaml, err := generateVendirSecretYaml(secretName, credentials.Username, credentials.Password)
```
Comment (fritzduchardt) [inline internal/myks/globe.go:447]:
```suggestion
		secretYaml, err := g.generateVendirSecretYaml(secretName, credentials.Username, credentials.Password)
```

File: internal/myks/sync.go [hunk 3]
```diff
@@ -124,65 +122,14 @@ func (a *Application) doSync() error {
 				log.Info().Msg(a.Msg(syncStepName, "Resource already synced"))
 				continue
 			}
-			args := []string{
```
Comment (fritzduchardt) [inline internal/myks/sync.go:127]: So much code obsolete - nice

File: internal/myks/util_test.go [hunk 4]
```diff
@@ -390,3 +392,22 @@ func TestProcess(t *testing.T) {
...
 	}
 }
+
+func assertEqual(t *testing.T, got, want interface{}) {
+	if !reflect.DeepEqual(got, want) {
+		t.Errorf("Expected:\n%v\nGot:\n%v\nDifference:\n%v", want, got, diff(got, want))
+	}
+}
+
+func diff(a, b interface{}) string {
```
Comment (fritzduchardt) [inline internal/myks/util_test.go:402]: Awesome addition

**Review by fritzduchardt — COMMENTED** [review]
Added some tweaks to the regex and logging.

File: internal/myks/util_test.go (hunk 4 above)
Comment (Zebradil) [reply internal/myks/util_test.go:402]: Generated by ChatGPT™ 🙃

File: internal/myks/sync.go (hunk 3 above)
Comment (Zebradil) [reply internal/myks/sync.go:127]: I remember we had a similar approach in the original `doer.sh` — two almost identical branches to sync directories one by one or all at once.

**Review by Zebradil — COMMENTED** [review]
@fritzduchardt thanks for the review!

**Review by fritzduchardt — APPROVED** [review]

**Pull request merged**